### ISSUE TITLE
If I want to scale the convex hull of a rigid body, what is the correct operation?

### ISSUE THREAD

**Issue by CeSun** [notgiven688/jitterphysics2#26]
Transform all vertices and reconstruct the shape?

**Comment by notgiven688** [comment]
> Transform all vertices and reconstruct the shape?

Yes, transforming all vertices and reconstructing the shape would be the method of choice. However, there's an exception: if you have many instances of convex hulls that only differ in their scaling factor. In such cases, you can wrap your `ConvexHullShape` inside a `TransformedShape`. To perform the scaling transformation, modify the diagonal entries of the "orientation" matrix in the constructor. I will rename the matrix, as it supports not only orientation but also scaling. In fact, all affine transformations (see [Affine Transformation](https://en.wikipedia.org/wiki/Affine_transformation)) are supported by `TransformedShape`.

Here's an example from JitterDemo, in ConvexDecomposition.cs:

Change:

```cs
foreach (ConvexHullShape s in shapesToAdd)
{
  body.AddShape(s.Clone(), false);
}
```

To:

```cs
foreach (ConvexHullShape s in shapesToAdd)
{
  body.AddShape(new TransformedShape(s.Clone(), JMatrix.Identity * 0.2f, JVector.Zero), false);
}
```

This will scale your object by a factor of 0.2.
